Dockerfile

```diff
@@ -0,0 +1,95 @@
+# Base container that is used for both building and running the app
+FROM fedora:30 as base
+ENV RUBY_MODULE="ruby:2.6"
+ENV NODEJS_MODULE="nodejs:11"
+
+RUN \
+  echo "tsflags=nodocs" >> /etc/dnf/dnf.conf && \
+  dnf -y upgrade && \
+  dnf -y module install ${RUBY_MODULE} ${NODEJS_MODULE} && \
+  dnf -y install mysql-libs mariadb-connector-c postgresql-libs ruby{,gems} rubygem-{rake,bundler} nc hostname && \
+  dnf clean all && \
+  rm -rf /var/cache/dnf/
+
+ENV HOME=/home/foreman
+WORKDIR $HOME
+RUN groupadd -r foreman -f -g 1001 && \
+    useradd -u 1001 -r -g foreman -d $HOME -s /sbin/nologin \
+    -c "Foreman Application User" foreman && \
+    chown -R 1001:1001 $HOME
+
+# Add a script to be executed every time the container starts.
+COPY entrypoint.sh /usr/bin/
+RUN chmod +x /usr/bin/entrypoint.sh
+ENTRYPOINT ["entrypoint.sh"]
+
+# Temp container that download gems/npms and compile assets etc
+FROM base as builder
+
+RUN \
+  dnf -y install redhat-rpm-config git \
+    gcc-c++ make bzip2 \
+    libxml2-devel libcurl-devel ruby-devel \
+    mysql-devel postgresql-devel libsq3-devel && \
+  dnf clean all && \
+  rm -rf /var/cache/dnf/
+
+ENV RAILS_ENV="production"
```

Comment:
I see that you use this ENV for both build-stages. Did you consider this: https://github.com/moby/moby/issues/37345#issuecomment-400245466?

All in all, it looks good to me and I believe the image size was reduced drastically.

Reply:
@boaz0 thanks, I did try that, but it seems the docker version that we have on quay.io doesnt support that (https://quay.io/repository/ohadlevy/foreman/build/395a5fa8-f543-4cae-8731-54a24acd9730).

I did do minor cleanups on the ARGS but they are still duplicated across the images.